test/graphql/type/union_interface_test.exs

```diff
@@ -0,0 +1,280 @@
+    query = """
+      {
+        __typename
+        name
+        pets {
+          __typename
+          ... on Dog {
+            name
+            barks
+          }
+          ... on Cat {
+            name
+            meows
+          }
+        }
+      }
+    """
+    assert_execute {query, schema, john},
+      %{"__typename" => "Person",
+        "name" => "John",
+        "pets" => [
+          %{"__typename" => "Dog", "barks" => true, "name" => "Odie"},
+          %{"__typename" => "Cat", "meows" => false, "name" => "Garfield"}
+        ]
+      }
+  end
+
+  test "executes using interface types" do
+    # NOTE: This is an *invalid* query, but it should be an *executable* query.
+    query = """
+      {
+        __typename
+        name
+        friends {
+          __typename
+          name
+          barks
+          meows
+        }
+      }
+    """
+
+    assert_execute {query, schema, john},
+      %{"__typename" => "Person",
+        "friends" => [
+          %{"__typename" => "Person", "name" => "Liz"},
+          %{"__typename" => "Dog", "barks" => true, "name" => "Odie"}
+        ],
+        "name" => "John"
+      }
+  end
+
+  test "executes types with inline fragments" do
+    # This is the valid version of the query in the above test.
+    query = """
+    {
+        __typename
+        name
+        friends {
+          __typename
+          name
+          ... on Dog {
+            barks
+          }
+          ... on Cat {
+            meows
+          }
+        }
+      }
+    """
+    assert_execute {query, schema, john},
+      %{"__typename" => "Person",
+        "friends" => [
+          %{"__typename" => "Person", "name" => "Liz"},
+          %{"__typename" => "Dog", "barks" => true, "name" => "Odie"}
+        ],
+        "name" => "John"
+      }
+  end
+
+  test "allows fragment conditions to be abstract types" do
+    query = """
+      {
+        __typename
+        name
+        pets { ...PetFields }
+        friends { ...FriendFields }
+      }
+
+      fragment PetFields on Pet {
+        __typename
+        ... on Dog {
+          name
+          barks
+        }
+        ... on Cat {
+          name
+          meows
+        }
+      }
+
+      fragment FriendFields on Named {
+        __typename
+        name
+        ... on Dog {
+          barks
+        }
+        ... on Cat {
+          meows
+        }
+      }
+      """
+      assert_execute {query, schema, john},
+        %{"__typename" => "Person",
+          "name" => "John",
+          "friends" => [
+            %{"__typename" => "Person", "name" => "Liz"},
+            %{"__typename" => "Dog", "barks" => true, "name" => "Odie"}
+          ],
+          "pets" => [
+            %{"__typename" => "Dog", "barks" => true, "name" => "Odie"},
+            %{"__typename" => "Cat", "meows" => false, "name" => "Garfield"}
+          ]
+        }
+  end
+
+  test "gets execution info in resolver" do
+
```

Comment:
Empty test here
Here's the reference from the JS implementation. https://github.com/graphql/graphql-js/blob/042e1620bb3a83e07e15586bb5b69b211c91e8de/src/execution/__tests__/union-interface.js#L350